src/native/eventpipe/ep-buffer.c

```diff
@@ -150,13 +150,7 @@ ep_buffer_move_next_read_event (EventPipeBuffer *buffer)
 			EP_ASSERT (!"Input pointer is out of range.");
 			buffer->current_read_event = NULL;
 		} else {
-			if (ep_event_instance_get_data (buffer->current_read_event))
-				// We have a pointer within the bounds of the buffer.
-				// Find the next event by skipping the current event with it's data payload immediately after the instance.
-				buffer->current_read_event = (EventPipeEventInstance *)ep_buffer_get_next_aligned_address (buffer, (uint8_t *)(ep_event_instance_get_data (buffer->current_read_event) + ep_event_instance_get_data_len (buffer->current_read_event)));
-			else
-				// In case we do not have a payload, the next instance is right after the current instance
-				buffer->current_read_event = (EventPipeEventInstance *)ep_buffer_get_next_aligned_address (buffer, (uint8_t *)(buffer->current_read_event + 1));
+			buffer->current_read_event = (EventPipeEventInstance *)ep_buffer_get_next_aligned_address (buffer, (uint8_t *)buffer->current_read_event + ep_event_instance_get_flattened_size (buffer->current_read_event));
```

Comment:
Feels like this changed behavior, the previous code didn't do any calculations related to stack size.

Reply:
It did implicitly. The old calculation was doing pointer arithmetic on `EventPipeEventInstance`s in the array. That `+1` is effectively a `(uint8_t*)(instance_array) + sizeof (EventPipeEventInstance)`. That struct used to have the fixed size `EventPipeStackContents` struct in it, so it would include the stack in the pointer movement.

Reply:
OK, I missed that we did pointer calculation on a EventPipeInstance *, so then its ep_event_instance_get_flattened_size should be doing the same thing.